src/HotTable.vue

```diff
@@ -86,6 +98,38 @@
 
         preventInternalEditWatch(this);
       },
+      matchHotMappersSize: function (data: any[][]): void {
+        const rowsToRemove: number[] = [];
+        const columnsToRemove: number[] = [];
+        const indexMapperRowCount = this.hotInstance.rowIndexMapper.getNumberOfIndexes();
+        const indexMapperColumnCount = this.hotInstance.columnIndexMapper.getNumberOfIndexes();
+
+        if (data && data.length !== indexMapperRowCount) {
+          if (data.length < indexMapperRowCount) {
+            for (let r = data.length; r < indexMapperRowCount; r++) {
+              rowsToRemove.push(r);
+            }
+
+            this.hotInstance.rowIndexMapper.removeIndexes(rowsToRemove);
+
+          } else {
+            this.hotInstance.rowIndexMapper.insertIndexes(indexMapperRowCount - 1, data.length - indexMapperRowCount);
+          }
+        }
+
+        if (data && data[0] && data[0]?.length !== indexMapperColumnCount) {
+          if (data[0].length < indexMapperColumnCount) {
+            for (let c = data[0].length; c < indexMapperColumnCount; c++) {
+              columnsToRemove.push(c);
+            }
+
+            this.hotInstance.columnIndexMapper.removeIndexes(columnsToRemove);
+
+          } else {
+            this.hotInstance.rowIndexMapper.insertIndexes(indexMapperColumnCount - 1, data[0].length - indexMapperColumnCount);
+          }
+        }
```

Comment:
Gather all changes to index mappers and run them in batch. Should result in a single render

```suggestion
        if (data && data.length !== indexMapperRowCount) {
          if (data.length < indexMapperRowCount) {
            for (let r = data.length; r < indexMapperRowCount; r++) {
              rowsToRemove.push(r);
            }
          }
        }
        
        this.hotInstance.batch(() => {
            if (rowsToRemove.length > 0) {
              this.hotInstance.rowIndexMapper.removeIndexes(rowsToRemove);
            } else {
              this.hotInstance.rowIndexMapper.insertIndexes(indexMapperRowCount - 1, data.length - indexMapperRowCount);
            }
            
            if (columnsToRemove.length > 0) {
              this.hotInstance.columnIndexMapper.removeIndexes(columnsToRemove);
            } else {
              this.hotInstance.rowIndexMapper.insertIndexes(indexMapperColumnCount - 1, data[0].length - indexMapperColumnCount;
            }
        });
```

Reply:
> Should result in a single render

@wojciechczerniak I can run them in a batch, but I don't think they trigger a render anyway? https://jsfiddle.net/js_ziggle/ndsg1906/

Reply:
Sorry, my bad. I was thinking about `updateCache` https://jsfiddle.net/mfp316rq/  But here we always have just one call for each indexMapper so I guess it doesn't matter.